internal/cloud/realm/dependencies.go

```diff
@@ -49,20 +83,43 @@ func (c *client) ImportDependencies(groupID, appID, uploadPath string) error {
 		return err
 	}
 
-	res, resErr := c.do(
-		http.MethodPost,
+	res, err := c.do(
+		http.MethodPut,
 		fmt.Sprintf(dependenciesPathPattern, groupID, appID),
 		api.RequestOptions{
 			Body:        body,
 			ContentType: w.FormDataContentType(),
 		},
 	)
-	if resErr != nil {
-		return resErr
+	if err != nil {
+		return err
 	}
 	if res.StatusCode != http.StatusNoContent {
 		return api.ErrUnexpectedStatusCode{"import dependencies", res.StatusCode}
 	}
+
+	waitForTranspilation := func() error {
+		status, err := c.GetDependenciesStatus(groupID, appID)
+		if err != nil {
+			return err
+		}
+		for status == DependenciesStatusCreated {
+			time.Sleep(time.Second)
+			status, err = c.GetDependenciesStatus(groupID, appID)
+			if err != nil {
+				return err
+			}
+			if status == DependenciesStatusFailed {
+				return fmt.Errorf("failed to install dependencies")
+			}
+		}
+		return nil
+	}
+
+	if err := waitForTranspilation(); err != nil {
```

Comment:
i may not be fully aware of the requirements on this, but I don't think this "Wait for" should live in the implementation of realm client's `ImportDependencies`.  i get the desire to make sure we're always waiting for the successful install (and report failures otherwise) whenever we call `ImportDependencies`, but so far in the CLI that operation is only called once (in the `push` command).

this logic can/should just live there for now.  if/when we ever need this to be abstracted away, it would then make sense to make a helper (similar to `cli.ResolveApp`) which encapsulated all of this.

the main reason this caught my eye is something andy already alluded to: using a spinner.  moreover, we should avoid placing any "ui logic" (or related package imports) in the `cloud/realm` package